### ISSUE TITLE
State management

### ISSUE THREAD

**Issue by gooftroop** [openventilatorregistry/Frontend#4]
Answer question after looking at the designs - do we need global state management or not? Sell us on it on why we do.
If we do, provide a recommendation for a library.

**Comment by chriswillsflannery** [comment]
My initial impression is that the only centralized state we would need would be auth data - could be localized and passed through context. Not sure I'm immediately seeing the need for something like Redux but maybe there's something I missed. Thoughts?

**Comment by tqmai** [comment]
Yeah, the app seems small enough that we wouldn't have that much prop drilling, but I'm also not opposed to using something like Redux for state management. Context API is another option that is lighter than Redux but provides centralized state.

**Comment by chriswillsflannery** [comment]
<img width="840" alt="Screenshot 2020-04-01 at 02 05 19" src="https://user-images.githubusercontent.com/6463453/78278101-8a01e680-74e3-11ea-8265-91cdc0a7f35f.png">
<img width="474" alt="Screenshot 2020-04-01 at 02 05 27" src="https://user-images.githubusercontent.com/6463453/78278103-8c644080-74e3-11ea-897b-11f928311d51.png">

Based on the design team's current information architecture still TBD whether we will need a centralized store of state for anything other than auth credentials - take for example, the 'create hospital profile' path -> this would be a form which would send send newly created hospital data to the DB, and update a useReducer hook local to the Hospital Profile Page component. Not immediately apparent whether that particular hospital's associated data would be needed anywhere else in the app tree other than the Hospital Profile Page's immediate children. Thoughts?

**Comment by tqmai** [comment]
If I'm understanding the design correctly, the _index of hospitals_ and _index of suppliers_ would require a full list of the hospitals/suppliers, so I imagine the easiest way to manage this is to store the data of all the hospitals together and all the suppliers together. If that's the case and if we don't have a centralized store of state, we would probably need to store the state in the Homepage component and do a decent amount of prop drilling to get the data to the other components. So given that, I could see the value of having a centralized store of state. So I think using Context could work well in this situation. What do you think?
